From c272e70fbc28a5a8ab26e9c949e8ac7830020705 Mon Sep 17 00:00:00 2001
From: Takashi Yano <takashi.yano@nifty.ne.jp>
Date: Thu, 5 May 2022 21:12:26 +0900
Subject: [PATCH 066/N] Cygwin: sigproc: Avoid segfault caused by signal just
 after fork().

- The commit "Cygwin: always add sigmask to child info" also tries
  to fix this issue, however, did not fix enough. This patch fixes
  that.

Signed-off-by: Johannes Schindelin <johannes.schindelin@gmx.de>
---
 winsup/cygwin/sigproc.cc | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/winsup/cygwin/sigproc.cc b/winsup/cygwin/sigproc.cc
index b5143e8..fbb1f0e 100644
--- a/winsup/cygwin/sigproc.cc
+++ b/winsup/cygwin/sigproc.cc
@@ -1376,9 +1376,9 @@ wait_sig (VOID *)
 	     when _main_tls points to the system-allocated stack, not to
 	     the parent thread. In that case find_tls fails, and we fetch
 	     the sigmask from the child_info passed from the parent. */
-	  tl_entry = cygheap->find_tls (_main_tls);
-	  if (tl_entry)
+	  if (cygwin_finished_initializing)
 	    {
+	      tl_entry = cygheap->find_tls (_main_tls);
 	      dummy_mask = _main_tls->sigmask;
 	      cygheap->unlock_tls (tl_entry);
 	    }
-- 
2.9.0

